neo/Helper.cs

```diff
@@ -320,5 +320,18 @@ public static IConfigurationRoot LoadConfig(string config)
                 .AddJsonFile(configFile, true)
                 .Build();
         }
+
+        public static T StrictEnum<T>(byte value) where T : Enum
+        {
+            T state = (T)Enum.ToObject(typeof(T), value);
+            if (!EnumValueCache<T>.definedValues.Contains(state))
+                throw new ArgumentOutOfRangeException();
```

Comment:
I think the exception should be thrown by the caller. Because we don't know what exception should be thrown. Otherwise, the caller should catch the exception the re-throw the correct one.

Reply:
uhh, so the feedback is
- drop this helper function
- type boiler plate `if (!Enum.IsDefined(typeof(something), enum_variable) raise component_specific_exception` in all places that the project did not do enum validation

correct?

Reply:
Yes.